docs/network-binding-plugin.md

```diff
@@ -0,0 +1,342 @@
+# Network Binding Plugin
+[v1.1.0, Alpha feature]
+
+A modular plugin framework which integrates with Kubevirt to implement a
+network binding.
+
+This document is aimed at plugin developers which would like to create
+a custom network binding and integrate it with Kubevirt.
+
+Before proceeding with the reading of this developer guide, please read
+the [Network Binding Plugins](https://kubevirt.io/user-guide/virtual_machines/network_binding_plugins/)
+user guide.
+
+## Overview
+
+### Network Connectivity
+In order for a VM to have access to external network(s), several layers
+need to be defined and configured, depending on the connectivity characteristics
+needs.
+
+These layers include:
+
+- Host connectivity: Network provider.
+- Host to Pod connectivity: CNI.
+- Pod to domain connectivity: Network Binding.
+
+This guide focuses on the Network Binding portion.
+
+### Network Binding
+The network binding defines how the domain (VM) network interface is wired
+in the VM pod through the domain to the guest.
+
+The network binding includes:
+
+- Domain vNIC configuration.
+- Pod network configuration (optional).
+- Services to deliver network details to the guest (optional).
+  E.g. DHCP server to pass the IP configuration to the guest.
+
+### Plugins
+The network bindings have been part of Kubevirt core API and codebase.
+With the increase of the number of network bindings added and
+frequent requests to tweak and change the existing network bindings,
+a decision has been made to create a network binding plugin infrastructure.
+
+The plugin infrastructure provides means to compose a network binding plugin
+and integrate it into Kubevirt in a modular manner.
+
+Kubevirt is providing several network binding plugins as references.
+The following plugins are available:
+
+- [passt](https://kubevirt.io/user-guide/virtual_machines/net_binding_plugins/passt/) [v1.1.0]
+- [macvtap](https://kubevirt.io/user-guide/virtual_machines/net_binding_plugins/macvtap/) [v1.1.1]
+- [slirp](https://kubevirt.io/user-guide/virtual_machines/net_binding_plugins/slirp/) [v1.1.0]
+
+## The Zero Code Plugin
```

Comment:
It seems the example in this section is for a scenario where we want to run a custom CNI to plumb the network inside the container.
I think its worth to mention it at the beginning.
Also, personally I dont know about zero-code-plugin term, I dont think most CNIs would work out of the box, aren't they?